### ISSUE TITLE
submodules

### ISSUE THREAD

**Issue by kadamski** [breadbee/breadbee_dev#5]
You seem to like to have this "bootstrap" target that clones and sets up some of the additional repositories. While submodules do have their problems, this seems to be the most common way of handling such a use-case. 

Do you have any specific reasons not to use this approach?

**Comment by mbilloo** [comment]
@kadamski , I agree. It looks like the Makefile is effectively doing the equivalent of submodules.

**Comment by fifteenhex** [comment]
Sorry, I didn't see this until now.

The reason to not use submodules here is that I have a bunch of different kernel branches I'm working on and that would probably cause git to constantly want to push references to commits that aren't on github.

**Comment by kadamski** [comment]
@fifteenhex I'm not sure I understand your concern. The submodule works similar to other files - unless you "git add" it, the change won't be part of the commit. So you can change the branch or do some commits in the submodule subdir and not push this with other changes.

There is a risk, however, that you would "git add" submodule but would forget to do a "git push" in the submodule git directory, of course. But I'm not sure if it is any worse than the current setup where you don't really use the "bootstrap" target so it is not always working anyways :) At least a partial solution for this problem would be to track branch in the submodule which is possible in newer git versions.

**Comment by fifteenhex** [comment]
> "git add" it, the change won't be part of the commit.

I'm the type of person that uses git commit -a. ;)
